Develop

wjones98 opened the pull request.

LydiaLane committed:
Add location field and add addItem route

LydiaLane committed:
query url to display add item parameters in stock table

u-blavins committed:
Add item API with model; Update basket api

LydiaLane committed:
parse stock into inventory table

u-blavins committed:
Merge pull request #13 from u-blavins/lydia_view_basket_branch

Parse inventory into stock table

u-blavins committed:
Update items model and route

u-blavins committed:
Update items

u-blavins committed:
Update add basket checking against items in db

LydiaLane committed:
add unit type to add basket page

u-blavins committed:
Merge pull request #14 from u-blavins/lydia_view_basket_branch

add unit type to add basket page

LydiaLane committed:
accept/deny users page

u-blavins committed:
Update web app with adding to basket

LydiaLane committed:
accept users page, css changes, registration fields edited

wjones98 committed:
Merge pull request #16 from u-blavins/lydia_view_basket_branch

stock page changes - accept/deny users

wjones98 committed:
Database stuff

wjones98 committed:
Merge branch 'develop' of https://github.com/u-blavins/aad_project into develop

wjones98 committed:
Model Method and Route for getting users waiting for approval

u-blavins committed:
Update template for basket to pull from items model

wjones98 committed:
/users/approval now displays users waiting for approval

u-blavins committed:
Update basket so that users can add and remove items

wjones98 committed:
Changed approve users to admin.py

LydiaLane committed:
smol change to rebase

wjones98 committed:
Merge pull request #17 from u-blavins/feature/customer-basket

Update basket so that users can add and remove items

LydiaLane committed:
smol

LydiaLane committed:
styles

LydiaLane committed:
Merge pull request #18 from u-blavins/lydia_view_basket_branch

Lydia branch

LydiaLane committed:
icons to navbar

N0689085 committed:
returnItems.html and Js

LydiaLane committed:
styling

LydiaLane committed:
Merge pull request #19 from u-blavins/lydia_view_basket_branch

styling

wjones98 committed:
approve/deny users feature implemented

wjones98 committed:
Merge branch 'develop' of https://github.com/u-blavins/aad_project into develop

LydiaLane committed:
responsive tables and text

u-blavins committed:
Update unit values

u-blavins committed:
Merge branch 'develop' of https://github.com/u-blavins/aad_project into develop

N0689085 committed:
return Items Development

N0689085 committed:
Merge branch 'develop' of https://github.com/u-blavins/aad_project into develop

u-blavins committed:
Merge branch 'develop' of https://github.com/u-blavins/aad_project into develop

u-blavins committed:
Update returnItems logic

LydiaLane committed:
Styling

LydiaLane committed:
styling

LydiaLane committed:
Merge pull request #20 from u-blavins/lydia_view_basket_branch

Styling/responsive elements

u-blavins committed:
Implement full functionality for basket; Update view stock page

wjones98 committed:
just fixing Usamas spelling errors :)

wjones98 committed:
Refactored Database Utility

wjones98 committed:
Merge pull request #23 from u-blavins/WAJ-Utils-DB-Feature

Borth clicked the button

u-blavins committed:
Update basket to order items and display a receipt

u-blavins committed:
Update customer with latest Database update

wjones98 committed:
Merge pull request #21 from u-blavins/feature/customer-basket-order

Customer Basket Functionality

u-blavins committed:
Add ec2 cloudformation template

u-blavins committed:
Update EC2 template

u-blavins committed:
Update template to be reusuable for anyone

LydiaLane committed:
Fix auth css

LydiaLane committed:
User account page

LydiaLane committed:
Merge pull request #25 from u-blavins/lydia_view_basket_branch

user settings page

LydiaLane committed:
styling

LydiaLane committed:
styling

LydiaLane committed:
Merge pull request #26 from u-blavins/lydia_view_basket_branch

styling

wjones98 committed:
Merge pull request #24 from u-blavins/feature/ec2-webserver

Add EC2 template for deployment

u-blavins committed:
Upate user account page; Add Transaction models

wjones98 committed:
Merge pull request #27 from u-blavins/feature/user-account

Upate user account page; Add Transaction models

LydiaLane committed:
add item to inventory

N0689085 committed:
error handling (staff department code)

N0689085 committed:
Merge branch 'develop' of https://github.com/u-blavins/aad_project into develop

wjones98 committed:
fixed register bug

Trying to register as staff was not passing departmentcode through resulting in them not being able to register. Although not pretty added an if + else into the route call that sets the department code to ST4FF if privelege above 2

LydiaLane committed:
Merge pull request #28 from u-blavins/lydia_view_basket_branch

add item to inventory feature

wjones98 committed:
Changes to transaction model

changed query to use a custom sql server function that takes user ID as paramater, same outcome but cleaner query. Also added order by to group items of the same transaction together as well as order by date desc

N0689085 committed:
password function (check matching) added

N0689085 committed:
Merge branch 'develop' of https://github.com/u-blavins/aad_project into develop

u-blavins committed:
Update userSettings.hmtl with error messages

u-blavins committed:
Fix bug in update user password

u-blavins committed:
Remove console logging

LydiaLane committed:
Show error message upon user adding an item to stock with an existing ID

LydiaLane committed:
Merge branch 'lydia_view_basket_branch' of https://github.com/u-blavins/aad_project into lydia_view_basket_branch

LydiaLane committed:
Change table row colour if the stock quantity is less than the min threshold

u-blavins committed:
Update user routes by removing User ID from view

LydiaLane committed:
Show and hide add/remove stock buttons based on user permissions

LydiaLane committed:
Remove items from stock

wjones98 committed:
Merge pull request #29 from u-blavins/feature/user-account

Update user accounts page

N0689085 committed:
Merge pull request #30 from u-blavins/lydia_view_basket_branch

Error msg for duplicate stock id and red table rows on view stock page if below threshold

u-blavins committed:
Updated auth validation

LydiaLane committed:
Added unit types to add item to stock

LydiaLane committed:
Format code to uppercase and add unit types to add basket page

LydiaLane committed:
PR changes

LydiaLane committed:
PR changes x2

u-blavins committed:
Merge pull request #31 from u-blavins/lydia_view_basket_branch

home page and add to basket changes

LydiaLane committed:
change password fix

LydiaLane committed:
Edit item in stock

N0689085 committed:
java, check for empty password

LydiaLane committed:
Show user friendly text instead of integers for perms requested

wjones98 committed:
View Purchase Orders Page Implemented

LydiaLane committed:
PR changes

u-blavins committed:
Update viewStock.html view to display for both Staff and Admin users

u-blavins committed:
Merge pull request #32 from u-blavins/lydia_view_basket_branch

Edit item in stock

u-blavins committed:
Update adding items to basket with privilege user check

wjones98 committed:
Implemented view purchase order functionality

wjones98 committed:
fixed html formatting

u-blavins committed:
Add functionality for admins to update user and staff view users

wjones98 committed:
Merge pull request #34 from u-blavins/feature/admin-update-user-privilege

Add functionality for admins to update user and staff view users

wjones98 committed:
Merge pull request #33 from u-blavins/feature/privilege-user-items

Update adding items to basket with privilege user check

u-blavins committed:
Update basket route to mitigate errors

N0689085 committed:
new returns page

wjones98 committed:
frontend formatting

N0689085 committed:
page setup(returnPage)

LydiaLane committed:
JS fix

u-blavins reviewed (commented):
Hold off on the PR since bugs within basket have been merged

N0689085 committed:
update

LydiaLane committed:
Add required to fields on add to basket

u-blavins committed:
Updated admin for returning an item

u-blavins committed:
Remove redundant files

u-blavins committed:
Merge pull request #37 from u-blavins/hotfix/add-items-to-basket

Update basket route to mitigate errors

wjones98 committed:
Confirming purchase order deliveries

u-blavins committed:
Udpate return items view with table

u-blavins committed:
Add purchase order table

u-blavins committed:
Update return item page

N0689085 committed:
added grey to .threads(css)

wjones98 committed:
Purchase Order Feature complete

wjones98 committed:
Client can now download csv file of purchase order

wjones98 committed:
Download csv now enabled for all purchase orders both pending and complete

wjones98 committed:
fixed bug with addPurchaseOrder.js referenced breaking add items page

wjones98 committed:
Merge branch 'develop' into WAJ-PurchaseOrder

u-blavins committed:
Merge pull request #38 from u-blavins/WAJ-PurchaseOrder

Waj purchase order

u-blavins committed:
Add view for returning items and working on creating model and logic

u-blavins committed:
Pull changes from purchase order

u-blavins committed:
Update functionality for refunds with error validation

u-blavins committed:
Update returning item with extra validation and refactoring

u-blavins committed:
Update required fields on table

u-blavins committed:
Update variable in returnItems.js

LydiaLane committed:
Merge pull request #39 from u-blavins/feature/return-items

Feature to refund an item

wjones98 committed:
download finance report for month/year implemented

wjones98 committed:
Added email button

N0689085 committed:
permission -> privilege (acceptUsers.html)

u-blavins committed:
Added Transaction model

wjones98 committed:
Ability to view costs for each department based on month/year

wjones98 committed:
Merge branch 'feature/finance-report' of https://github.com/u-blavins/aad_project into feature/finance-report

u-blavins committed:
Add functionality to email finance report

wjones98 committed:
view department transactions route

wjones98 committed:
Merge branch 'feature/finance-report' of https://github.com/u-blavins/aad_project into feature/finance-report

wjones98 committed:
implemented full view billing featuer functionality

Staff can now go from the department all the way down to the transaction info

wjones98 committed:
added missing div

N0689085 committed:
Merge pull request #40 from u-blavins/feature/finance-report

Feature/finance report

wjones98 committed:
Frontend visual improvements

wjones98 committed:
initial scanning code

u-blavins committed:
Update api routes to stop info being made public

wjones98 committed:
FIXED IT

u-blavins committed:
Hotfix update adding item to stock issue

wjones98 committed:
working barcode scanner

bow down to your new overlord, big man waj

wjones98 committed:
attempted barcode bugfix

wjones98 committed:
Bug fixed (error not with code)

wjones98 committed:
Fully functioning barcode scanner implementation

check out the spicy css, honestly lads expecting a 5 on the contribution for this one

wjones98 committed:
barcode scanner css styling

wjones98 committed:
mobile test

wjones98 committed:
Revert "mobile test"

This reverts commit 31a052856c0a901ad7ae956e398dc6fccd0c4109.

wjones98 committed:
Revert "barcode scanner css styling"

This reverts commit d5a38e5428f7bdf91e9a8f4fc8595cb7fafb7aff.

wjones98 committed:
nice pink x button for overlay and barcode scanner icon

wjones98 committed:
Fixed transaction info error on my accounts page

u-blavins committed:
Merge pull request #46 from u-blavins/hotfix-myaccount

Fixed transaction info error on my accounts page

wjones98 committed:
colourful icons :)

u-blavins committed:
Merge branch 'develop' into feature/barcode-scanner

wjones98 committed:
Changes due to suggestions from Usama

u-blavins committed:
Merge pull request #47 from u-blavins/feature/barcode-scanner

Feature/barcode scanner

wjones98 committed:
fixed /transaction route bug

u-blavins committed:
Merge branch 'feature/barcode-scanner' into develop

wjones98 committed:
oopsies
